v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,125 @@
+const sha256 = require("sha256");
+const { BASE_ENDPOINT, ENDPOINTS } = require("./config");
+const {
+  defaultRequestConfig,
+  CustomError,
+  isDefinedAndNotNullAndNotEmpty,
+  defaultPutRequestConfig,
+  removeUndefinedAndNullValues
+} = require("../../util");
+
+const populateIdentifiers = (attributeArray, { Config }) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  let listType;
+  const AUDIENCE_LIST = ["email", "deviceId", "ipAddress"];
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array
+    attributeArray.forEach(element => {
+      if (AUDIENCE_LIST.includes(audienceType)) {
+        listType = audienceType;
+      }
+      if (!listType) {
```

Comment:
lets put the error in else rather than another if

Reply:
updated